libs/core/src/lib/bar/bar.component.html

```diff
@@ -0,0 +1,12 @@
+<div class="fd-bar"
+    [ngClass]="
+        (cosy ? ' fd-bar--cosy' : '') + 
+        (barDesign ? ' fd-bar--' + barDesign : '') + 
+        (inPage && !size ? ' fd-bar--page': '') + 
+        (inPage && size ? ' fd-bar--page-' + size: '') + 
+        (inHomePage && !size ? ' fd-bar--home-page': '') + 
+        (inHomePage && size ? ' fd-bar--home-page-' + size: '')">
```

Comment:
Could you consider using `applyCssClass` decorator for that case? There is a lot of statements.

Reply:
```typescript
    @applyCssClass
    /** CssClassBuilder interface implementation
     * function must return single string
     * function is responsible for order which css classes are applied
     */
    buildComponentCssClass(): string {
        return [
            'fd-bar',
            this.cosy ? 'fd-bar--cosy' : '',
           this.barDeisgn ? ('fd-bar--' + this.badDesign) : ''
            ....
        ].filter(x => x !== '').join(' ');
    }
```

Reply:
elementRef(): ElementRef<any> {
        return this.barElement;
    }